src/dashboard/user-interface/site-kit-widget-permanent-dismissal-route.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+namespace Yoast\WP\SEO\Dashboard\User_Interface;
+
+use Exception;
+use WP_Error;
+use WP_REST_Request;
+use WP_REST_Response;
+use Yoast\WP\SEO\Conditionals\No_Conditionals;
+use Yoast\WP\SEO\Dashboard\Infrastructure\Permanently_Dismissed_Site_Kit_Widget_Repository_Interface;
+use Yoast\WP\SEO\Main;
+use Yoast\WP\SEO\Routes\Route_Interface;
+
+/**
+ * Registers a route to set whether the Site Kit widget is permanently dismissed.
+ *
+ * @makePublic
+ *
+ * @phpcs:disable Yoast.NamingConventions.ObjectNameDepth.MaxExceeded
+ */
+class Site_Kit_Widget_Permanent_Dismissal_Route implements Route_Interface {
```

Comment:
I am not sure if we want to call the connection thing a `widget` in normal speech and in the code since it does not behave as a `widget` in the sense of widgets can be moved around

Reply:
Maybe we can just call it site_kit_configuration.

Reply:
Done!